client/src/inside/components/EhrContextBanner.vue

```diff
@@ -45,18 +52,39 @@ export default {
     },
     showSeeding () {
       return StoreHelper.isDevelopingContent()
+    },
+    isReadOnlyInstructor () {
+      return StoreHelper.isReadOnlyInstructor()
+    }
+  },
+  watch: {
+    '$route.query': (n) => {
+      if (n.readonly) {
+        this.setReadOnlyInstructor(true)
+      }
     }
   },
+
   methods: {
     toggleShow: function () {
       this.show = !this.show
       this.indicator = this.show ? '-' : '+'
+    },
+    setReadOnlyInstructor: function (isReadOnly) {
+      StoreHelper.setIsReadOnlyInstructor(isReadOnly)
     }
   },
   mounted: function () {
     if (StoreHelper.isInstructor(this)){
       this.show = true
     }
+    const { readonly } = this.$route.query
+    if (readonly) {
+      this.setReadOnlyInstructor(true)
+    }
+  },
+  beforeDestroy: function () {
+    this.setReadOnlyInstructor(true)
```

Comment:
I think it is the opposite. I have fixed it. But the proper behaviour is to unset it when the before the component is destroyed. This is a way to avoid unpredicted behaviour, and it renders the viewOnly view instead of the instructor editing view.